### PR TITLE
Fix paths for storyboard animation sprites

### PR DESCRIPTION
**Pull request by cadon0**

### CODE REVIEW

File: osu.Game/Storyboards/Drawables/DrawableStoryboardAnimation.cs [hunk 1]
```diff
@@ -118,8 +118,24 @@ private void load(TextureStore textureStore, Storyboard storyboard)
             for (int frame = 0; frame < Animation.FrameCount; frame++)
             {
                 string framePath = Animation.Path.Replace(".", frame + ".");
+                Drawable sprite = storyboard.CreateSpriteFromResourcePath(framePath, textureStore);
 
-                AddFrame(storyboard.CreateSpriteFromResourcePath(framePath, textureStore), Animation.FrameDelay);
+                if (sprite != null)
```
Comment (cadon0) [inline osu.Game/Storyboards/Drawables/DrawableStoryboardAnimation.cs]: The "extra frames without matching sprites" crash is resolved by the null checks

File: osu.Game/Storyboards/Drawables/DrawableStoryboardAnimation.cs [hunk 2]
```diff
@@ -118,8 +118,24 @@ private void load(TextureStore textureStore, Storyboard storyboard)
...
+                {
+                    AddFrame(sprite, Animation.FrameDelay);
+                    continue;
+                }
+
+                framePath = Animation.Path.Replace("0.", frame + ".");
+                sprite = storyboard.CreateSpriteFromResourcePath(framePath, textureStore);
```
Comment (cadon0) [inline osu.Game/Storyboards/Drawables/DrawableStoryboardAnimation.cs]: And the "first sprite with no appended 0" crash is resolved by checking for that scenario if the previous check did not find a sprite

File: osu.Game/Storyboards/Drawables/DrawableStoryboardAnimation.cs [hunk 3]
```diff
@@ -118,8 +118,24 @@ private void load(TextureStore textureStore, Storyboard storyboard)
...
+
+                if (sprite != null)
+                {
+                    AddFrame(sprite, Animation.FrameDelay);
+                }
+
+                // todo: handle animation intentionally declared with more frames than sprites to cause a blinking effect
+                // e.g. beatmap 5381's "spr\play-skip.png"
```
Comment (cadon0) [inline osu.Game/Storyboards/Drawables/DrawableStoryboardAnimation.cs]: And the blinking effect will likely be its own issue

File: osu.Game/Storyboards/Drawables/DrawableStoryboardAnimation.cs (hunk 2 above)
Comment (bdach) [reply osu.Game/Storyboards/Drawables/DrawableStoryboardAnimation.cs]: what's the point of the second replace? can't it just be `Animation.Path`?

File: osu.Game/Storyboards/Drawables/DrawableStoryboardAnimation.cs (hunk 3 above)
Comment (bdach) [reply osu.Game/Storyboards/Drawables/DrawableStoryboardAnimation.cs]: could probably be handled by making the animation a `DrawableAnimation` and adding empty drawables (or maybe even empty sprites) as frames

**Comment by cadon0** [comment]
Closing to open against ppy/osu

**Pull request closed without merging**